specification/sections/02-providers.md

```diff
@@ -150,3 +151,85 @@ class MyProvider implements Provider {
+
+#### Requirement 2.4.1
+
+> The `provider` interface **MUST** define an optional `initialize` function, which accepts the global `evaluation context` as an argument and performs initialization logic relevant to the provider.
+
+Many feature flag frameworks or SDKs require some initialization before they can be used.
+They might require the completion of an HTTP request, establishing persistent connections, or starting timers or threads.
+The `initialization` function is an ideal place for such logic.
+
+```java
+// MyProvider implementation of the initialize function defined in Provider
+class MyProvider implements Provider {
+  //...
+
+  // the global context is passed to the initialization function
+  void initialize(EvaluationContext initialContext) {
+    /*
+      A hypothetical initialization function: connect to a flag database, then pass a callback which fires PROVIDER_READY
+    */
+    this.connectToDatabase(() => this.emitEvent(ProviderEvents.READY));
+  }
+
+  //...
+}
+```
+
+### 2.5. Provider context reconciliation
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+Static-context focused providers may need a mechanism to understand when their cache of evaluated flags must be invalidated or updated. An `on-context-set` handler can be defined which performs whatever operations are needed to reconcile the evaluated flags with the new context.
+
+#### Requirement 2.5.1
+
+> The provider interface **MUST** define an `on context set` handler, which takes an argument for the previous context, and the newly set context, and which can be optionally implemented to reconcile any stored state pertaining to the global evaluation context.
+
+Especially in static-context implementations, providers and underlying SDKs my maintain a cache of evaluated flags for a particular context.
+The `on context set` handler provides a mechanism to update this state, often by re-evaluating flags in bulk with respect to the new context.
+
+```java
+// MyProvider implementation of the onContextSet function defined in Provider
+class MyProvider implements Provider {
+  //...
+
+  onContextSet(EvaluationContext oldContext, EvaluationContext newContext): void {
+    // update context-sensitive cached flags, or otherwise react to the change in the global context 
+  }
+
+  //...
+}
+```
+
+see: [static-context paradigm](../glossary.md#static-context-paradigm)
+
+### 2.6. Shutdown
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Requirement 2.6.1
+
+> The provider interface **MUST** define a `shutdown` or `dispose` function, which can be optionally implemented to perform whatever cleanup or shutdown is necessary for the provider implementation.
```

Comment:
Why **must**? Why not make it optional? If the provider doesn't need to do anything on shutdown, why not have a default (no-op) method in the interface so the provider doesn't have to do anything?

It's not hugely important, as implementing a no-op method is trivial anyway. I'm just curious as to why it's mandatory.